### DISCUSSION TITLE
incorrect results with constraint bounds containing unconstrained solution

### DISCUSSION THREAD

**Discussion by markus-grabner** [coin-or/Ipopt#748] [Q&A] [answered]
Hi, I'm trying to solve a rather simple problem with Ipopt-3.14.14 (quadratic objective in 4 variables, 4 linear constraints). When the constraints are omitted, Ipopt finds a reasonable solution, even satisfying the (yet disabled) constraints. However, as soon as the constraints are added (such that the unconstrained solution is contained within the bounds), Ipopt reports a solution where 2 of the constraint bounds are attained exactly, which is incorrect since it gives a much larger value of the objective function. It seems that the constraint bounds somehow "attract" the solution instead of just defining the allowed search space. The provided derivatives are checked successfully by the built-in derivative checker.

Can someone please give me a hint which option might affect this behaviour, and which additional information is required to further diagnose the problem (e.g., which part of the Ipopt console output)?

Thanks & kind regards,
Markus

**Comment by svigerske** [comment]
Maybe the constraint left/right hand side (`g_L, g_U`) are set incorrectly, so they become equality constraints instead of inequalities?

**Reply by markus-grabner** [reply, marked as answer]
Thanks for your time and your suggestion! Actually it was my fault, I misinterpreted the "new_x" flag. Strangely, this only affected the results when constraints were used, therefore I assumed that I didn't setup the constraints correctly. After fixing this, there are currently no obstacles to advancing to relevant problem sizes...

Kind regards,
Markus
